src/components/core/GeoChart.tsx

```diff
@@ -41,28 +56,53 @@ export type IGeoChartInnerProps = ICoreGeoChartProps &
     IDataSourceProviderInjectedProps &
     WrappedComponentProps;
 
+export interface IGeoChartInnerState {
+    enabledLegendItems: IPushpinCategoryLegendItem[];
+}
+export interface IGeoChartInnerOptions {
+    geoData: IGeoData;
+    segmentData: IPushpinCategoryLegendItem[];
+    sizeData: number[];
+}
 /**
  * Geo Chart react component
  */
-export class GeoChartInner extends BaseVisualization<IGeoChartInnerProps, {}> {
+export class GeoChartInner extends BaseVisualization<IGeoChartInnerProps, IGeoChartInnerState> {
     public static defaultProps: Partial<IGeoChartInnerProps> = {
         ...commonDefaultProps,
         chartRenderer: renderChart,
         legendRenderer: renderLegend,
-        legendPostion: TOP,
+        legendPosition: TOP,
         config: {
             mapboxToken: "",
         },
     };
 
+    private geoChartOptions: IGeoChartInnerOptions;
+
     public componentDidUpdate(prevProps: IGeoChartInnerProps) {
-        if (!isEqual(this.props.execution, prevProps.execution)) {
-            this.validateData();
+        if (isEqual(prevProps.execution, this.props.execution)) {
```

Comment:
I suggest  we just compare `prop.execution. executionResponse.dimensions` to improve perf since whole `execution` is huge object with large data.

Reply:
fixed